src/server/pps/server/api_server.go

```diff
@@ -760,6 +767,16 @@ func (a *apiServer) listJob(
 	jobs := a.jobs.ReadOnly(ctx)
 	jobInfo := &pps.JobInfo{}
 	_f := func(string) error {
+		if number == 0 {
+			return errutil.ErrBreak
+		}
+		if paginationMarker != nil {
+			createdAt := time.Unix(int64(jobInfo.Created.GetSeconds()), int64(jobInfo.Created.GetNanos())).UTC()
+			fromTime := time.Unix(int64(paginationMarker.GetSeconds()), int64(paginationMarker.GetNanos())).UTC()
+			if createdAt.Equal(fromTime) || !reverse && createdAt.After(fromTime) || reverse && createdAt.Before(fromTime) {
+				return nil
```

Comment:
should this return errutil.ErrBreak?

Reply:
hmm, I think you're right. errutil.ErrBreak will probably save some iterations over records that we know don't need to be included